src/Lexer.js

```diff
@@ -135,6 +134,15 @@ export class Lexer {
     if (this.options.pedantic) {
       src = src.replace(/^ +$/gm, '');
     }
+
+    if (this.options.gfm || this.options.pedantic) {
```

Comment:
gfm should act for the most part like commonmark.

```suggestion
    if (this.options.pedantic) {
```

Reply:
That causes a test failure, I'll see if I can't suss out how to address it:

```
Failures:
1) New html_comments should pass
  Message:
    Expected: Example 10</h3><!-- multi
    line    
    comment
    --><h3 id="exampl
      Actual: Example 10</h3><!-- multi
    line	
    comment
    --><h3 id="example-1
```

Reply:
It looks like you will have to change https://github.com/markedjs/marked/blob/master/test/specs/new/html_comments.html#L40

replace the spaces at the end of the line with a tab

Reply:
Looks like there was a tab character at the end of the multiline comment in `new/html_comments.md`, and the test was expecting it to be converted to 4 spaces. With this change in behavior, I believe the expected output (`html_comments.html`) should be updated to expect that the tab character be preserved instead since it's not a leading tab.